HomeWork5

rickdemonn opened the pull request:
HW5

*add test to DoctorApi
*refactoring repo
*refactoring controller
*add DTO
*change method GET /doctors - mayBeParams

Commit:
HomeWork5

Commit:
add plugin MupStruct

mfarsikov commented on src/main/java/hillel/spring/DoctorRestAPI/dto/DoctorDtoConverter.java:
```diff
@@ -0,0 +1,17 @@
+    @Mapping(target = "id", ignore = true)
+    Doctor toModel(DoctorInputDto dto);
+
+    DoctorOutputDto toModel(Doctor doc);
```
`toDto`

mfarsikov commented on src/main/java/hillel/spring/DoctorRestAPI/dto/DoctorDtoConverter.java:
```diff
@@ -0,0 +1,17 @@
+
+    DoctorOutputDto toModel(Doctor doc);
+
+    List<DoctorOutputDto> toModel(List<Doctor> docs);
```
If you have `Model` -> `DTO` function it is not the bests idea to have yet another `List<Model>`->`List<DTO>` one. because List (through stream) is a Functor and you always can apply `dtos.stream().map(::toModel).collect(toList())` and achieve the same functionality. 
Also, this approach works with other collections as Sets.

mfarsikov commented on src/test/java/hillel/spring/DoctorRestAPI/DoctorRestAPIControllerTest.java:
```diff
@@ -0,0 +1,141 @@
+                        "    \"id\": 1,\n" +
+                        "    \"name\": \"Aaa\",\n" +
+                        "    \"specialization\": \"surgeon\"\n" +
+                        "  }"));
```
I would recommend moving this JSON to .json file since escaped JSON is not readable at all

Commit:
Fix

*DoctorConverter refactoring
*Add .json file for tests
*In Controller disable injection DtoConvertor becouse dont work plugin on my work
